docs/modules/maintain-cluster/pages/enterprise-rest-api.adoc

```diff
@@ -0,0 +1,411 @@
+= ENTERPRISE REST API
+[[enterprise-rest-api]]
+
+Hazelcast Enterprise provides a REST API that allows you to access your data
+structures and cluster using the HTTP/HTTPS protocols.
+
+You can interact with the API using various tools and platforms such as cURL, REST clients (like Postman and Insomnia), or programming languages with HTTP request libraries or built-in support.
+
+The REST API also includes a Swagger page available at http://<host>:<port>/swagger-ui/index.html for making requests.
+
+NOTE: In this document, all the request examples are demonstrated using cURL.
+
+== Enabling REST API
+
+REST service is **disabled** in the configuration by default.
+You should enable it on your rest configuration as shown in the below declarative example:
+
+[tabs] 
+==== 
+XML:: 
++ 
+-- 
+[source,xml]
+----
+<hazelcast>
+    <rest enabled="true">
+        ...
+    </rest>
+</hazelcast>
+----
+--
+
+YAML::
++
+[source,yaml]
+----
+hazelcast:
+  rest:
+    enabled: true
+----
+====
+
+== Changing Default Port
+
+`8443` is the default port for REST API. To change it, you can use the rest configuration as shown in the below declarative example:
+
+[tabs] 
+==== 
+XML:: 
++ 
+-- 
+[source,xml]
+----
+<hazelcast>
+    <rest enabled="true">
+        <port>8080</port>
+    </rest>
+</hazelcast>
+----
+--
+
+YAML::
++
+[source,yaml]
+----
+hazelcast:
+  rest:
+    enabled: true
+    port: 8080
+----
+====
+
+== Authentication
+
+REST supports simple authentication. To configure authentication in the REST API, you should complete the following steps:
+
+* **#Enable Security:#**
+You need to enable the security to use REST API.
+
+* **#Configure Rest Realm:#**
+You need to introduce a REST realm as the authentication mechanism. This realm will handle user authentication using a simple method based on username and password.
+
+* **#Setup Users and Roles:#**
+You should identify users and their roles within the REST realm.
+
+* **#Configure AccessControlService:#**
+To configure `AccessControlService`, you should provide a factory class that creates an instance of the service. `AccessControlService` delegates authentication to the configured security realm. You can either create a custom factory class, or use the default implementation.
+For the default implementation, you can use `DefaultAccessControlServiceFactory`, to produce `DefaultAccessControlService`. Authorization decisions in this service are based on three predefined roles: `reader`, `writer` and `admin`.
+
+Following declarative example includes all the steps:
```

Comment:
Can you also describe similar to this:

The rest client needs to obtain a token based on this authentication. The token request will include the username and password as below and the client will get a token for one of the roles. For the following requests this token will be used until it expires.

Reply:
I separated Authentication part in 2 sections and explained it in second section. Could you please take a look at it?